### PR TITLE
Place Emoji fallback font at the front of the list

### PR DESCRIPTION
**Pull request by hterkelsen**
The symbols font has glyphs for common emojis, but they are very ugly compared to the emoji font glyphs. This PR causes us to always place the emoji font at the front of the fallback font list so we prioritize the better looking glyphs.

Fixes https://github.com/flutter/flutter/issues/86204

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: lib/web_ui/test/canvaskit/fallback_fonts_golden_test.dart [hunk 1]
```diff
@@ -123,6 +123,102 @@ void testMain() {
       // TODO: https://github.com/flutter/flutter/issues/71520
     }, skip: isIosSafari || isFirefox);
 
+    test('will put the Noto Emoji font before other fallback fonts in the list',
+        () async {
```
Comment (yjbanov) [inline lib/web_ui/test/canvaskit/fallback_fonts_golden_test.dart:127]: Should this test call `FontFallbackData.debugReset` before everthing else?

File: lib/web_ui/test/canvaskit/fallback_fonts_golden_test.dart [hunk 2]
```diff
@@ -123,6 +123,102 @@ void testMain() {
...
+      Completer<void> fontChangeCompleter = Completer<void>();
+      // Intercept the system font change message.
+      ui.window.onPlatformMessage = (String name, ByteData? data,
```
Comment (yjbanov) [inline lib/web_ui/test/canvaskit/fallback_fonts_golden_test.dart:130]: This seems a bit complicated. Can we use `notoDownloadQueue.debugWhenIdle()` instead? Maybe we can also use `isPending` and `debugIsLoadingFonts` to make sure the download has been initiated.

File: lib/web_ui/test/canvaskit/fallback_fonts_golden_test.dart (hunk 1 above)
Comment (hterkelsen) [reply lib/web_ui/test/canvaskit/fallback_fonts_golden_test.dart:127]: We already have `FontFallbackData.debugReset` in the `setUp`

File: lib/web_ui/test/canvaskit/fallback_fonts_golden_test.dart (hunk 2 above)
Comment (hterkelsen) [reply lib/web_ui/test/canvaskit/fallback_fonts_golden_test.dart]: Done. This cleans up a lot of copy-pasted code in this test

**Review by yjbanov — APPROVED** [review]
lgtm

**Pull request merged**